pig-tests/type-domains/sample-universe.ion

```diff
@@ -129,6 +129,7 @@
         (product product_to_remove whatever::symbol)
         (record record_to_remove (irrelevant int))
         (sum sum_to_remove (doesnt) (matter))
+        (sum will_be_replaced_with_product (will_be_replaced_with_product_variant t::product_to_remove))
```

Comment:
Not too big a fan of this type's name. Could it be renamed to start with `product` to be more consistent with the preceding types? Something like `product_to_be_replaced`? Also could remove the "will_be_replaced_with_product_" part before "variant".

Consider adding a comment saying this will be replaced in the permuted domain with another product of the same name.

Reply:
It's a `sum` type that will be replaced with a `product`.